source/common/tcp_proxy/upstream.h

```diff
@@ -74,8 +310,19 @@ class HttpConnPool : public GenericConnPool, public Http::ConnectionPool::Callba
     virtual ~Callbacks() = default;
     virtual void onSuccess(Http::RequestEncoder& request_encoder) {
       ASSERT(conn_pool_ != nullptr);
-      conn_pool_->onGenericPoolReady(host_, request_encoder.getStream().connectionInfoProvider(),
-                                     ssl_info_);
+      if (!Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.upstream_http_filters_with_tcp_proxy")) {
+        conn_pool_->onGenericPoolReady(host_, request_encoder.getStream().connectionInfoProvider(),
+                                       ssl_info_);
+        return;
+      }
+
+      Network::ConnectionInfoProviderSharedPtr local_connection_info_provider(
```

Comment:
want to pass `nullptr` but since `onGenericPoolReady()` accepts a reference, passing nullptr gives warning. I was not sure if I should change the function signature to accept a pointer. For now passing this dummy `local_connection_info_provider` which eventually gets ignored based on feature flag